virtualizarr/backend.py

```diff
@@ -39,6 +40,8 @@ class FileType(AutoName):
     zarr = auto()
     dmrpp = auto()
     zarr_v3 = auto()
+    kerchunk_json = auto()
+    kerchunk_parquet = auto()
```

Comment:
Could these be amalgamated into one reader? The "kerchunk" reader for the "kerchunk" references filetype? Then the reader itself detects if it's looking at a json or parquet?

Reply:
Good idea!

Reply:
Wondering if you have ideas on the auto-detection. Initially I was thinking of using the 'magic bytes' strategy that is done in `_automatically_determine_filetype`, but the Kerchunk storage format isn't really parquet, but more like a zarr mimicking directory structure containing multiple parquets. 

```
tree ref.parquet
├── air
│   └── refs.0.parq
├── lat
│   └── refs.0.parq
├── lon
│   └── refs.0.parq
└── time
    └── refs.0.parq
```

Reply:
I'm not sure how much we can do here. In general we need to put a bit more thought into whether or not the auto-detection can generalize for when we have a system of externally-defined readers plugging into a virtualizarr entrypoint. I feel like at best we're still going to have a detection system that is often ambiguous...